bot/cogs/tags.py

```diff
@@ -90,27 +90,37 @@ def _get_suggestions(self, tag_name: str, thresholds: Optional[List[int]] = None
         """
         Search for tags via contents.
 
-        `predicate` will be either any or all, or a custom callable to search. Must return a bool.
+        `predicate` will be the built-in any, all, or a custom callable. Must return a bool.
         """
         await self._get_tags()
 
-        keywords_processed: Tuple[str] = tuple(query.strip().casefold() for query in keywords.split(',') if query)
-        keywords_processed = keywords_processed or (keywords,)
-        founds: list = [
-            tag
-            for tag in self._cache.values()
-            if check(query in tag['embed']['description'] for query in keywords_processed)
-        ]
-
-        if not founds:
+        keywords_processed: List[str] = []
+        for keyword in keywords.split(','):
+            keyword_sanitized = keyword.strip().casefold()
+            if not keyword_sanitized:
+                # this happens when there are leading / trailing / consecutive comma.
+                continue
+            keywords_processed.append(keyword_sanitized)
+
+        if not keywords_processed:
+            # after sanitizing, we can end up with an empty list, for example when keywords is ','
+            # in that case, we simply want to search for such keywords directly instead.
+            keywords_processed = [keywords]
+
+        matching_tags = []
+        for tag in self._cache.values():
+            if check(query in tag['embed']['description'].casefold() for query in keywords_processed):
+                matching_tags.append(tag)
+
+        if not matching_tags:
             return None
-        elif len(founds) == 1:
-            return Embed().from_dict(founds[0]['embed'])
+        elif len(matching_tags) == 1:
+            return Embed().from_dict(matching_tags[0]['embed'])
         else:
-            is_plural: bool = len(keywords_processed) > 1 or any(kw.count(' ') for kw in keywords_processed)
+            is_plural = len(keywords_processed) > 1 or keywords.strip().count(' ') > 1
```

Comment:
I think this should be 0, not 1. Otherwise, something like "word1 word2" will be singular and only at "word1 word2 word3" will it become plural.

```suggestion
            is_plural = len(keywords_processed) > 1 or keywords.strip().count(' ') > 0
```

Reply:
Oh you are right, originally I wanted it to be just
```py
is_plural = len(keywords_processed) > 1 or keywords.strip().count(' ')
```
Thank you for the catch!